pkg/services/grpcserver/service.go

```diff
@@ -47,15 +48,28 @@ func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles, authe
 		enabled: features.IsEnabledGlobally(featuremgmt.FlagGrpcServer),
 	}
 
+	histogramOptions := prometheus.HistogramOpts{
+		Namespace: "grafana",
+		Name:      "grpc_request_duration_seconds",
+		Help:      "Time (in seconds) spent serving HTTP requests.",
+		Buckets:   instrument.DefBuckets,
+	}
+
+	if features.IsEnabledGlobally(featuremgmt.FlagEnableNativeHTTPHistogram) {
+		// the recommended default value from the prom_client
+		// https://github.com/prometheus/client_golang/blob/main/prometheus/histogram.go#L411
+		// Giving this variable an value means the client will expose the histograms as an
+		// native histogram instead of normal a normal histogram.
+		histogramOptions.NativeHistogramBucketFactor = 1.1
+		// The default value in OTel. It probably good enough for us as well.
+		histogramOptions.NativeHistogramMaxBucketNumber = 160
+		histogramOptions.NativeHistogramMinResetDuration = time.Hour
+	}
```

Comment:
it's probably worth moving this inside the `if grpcRequestDuration == nil {` block since it's not needed outside